updated_streamflow.py

```diff
@@ -0,0 +1,275 @@
+val_end = pd.Timestamp(val_end)
+test_start = pd.Timestamp(test_start)
+test_end = pd.Timestamp(test_end)
+
+df_plvo_merge['datetime'] = pd.to_datetime(df_plvo_merge['datetime'])
+df_plvo_train = df_plvo_merge[(df_plvo_merge['datetime'] >= train_start) & (df_plvo_merge['datetime'] <= train_end)]
+df_plvo_val = df_plvo_merge[(df_plvo_merge['datetime'] >= val_start) & (df_plvo_merge['datetime'] <= val_end)]
+df_plvo_test = df_plvo_merge[(df_plvo_merge['datetime'] >= test_start) & (df_plvo_merge['datetime'] <= test_end)]
+
+df_plvo_train.head()
+
+# Verify lengths of each set
+print("Training data length:", len(df_plvo_train))
+print("Validation data length:", len(df_plvo_val))
+print("Testing data length:", len(df_plvo_test))
+
+df_plvo_train.head()
+
+df_plvo_merge.shape
+
+# Drop columns except 'Sim_Discharge_Standardized' and 'USGS_Discharge'
+df_plvo_merged = df_plvo_merge[['Sim_Discharge_Standardized', 'USGS_Discharge_Standardized']]
+
+# Display the first few rows of the modified DataFrame
+df_plvo_merged.head()
+
+"""**Feature Engineering**"""
+
+# Define a function to prepare the dataset for LSTM
+def prepare_dataset(X_data, y_data, n_steps):
+    df_as_np = X_data.to_numpy()
+    df2_as_np = y_data.to_numpy()
+    X, y = [], []
+    for i in range(len(X_data) - n_steps):
+        row = [[a] for a in df_as_np[i:i+n_steps]]
+        X.append(row)
+        label = df2_as_np[i+n_steps]
+        y.append(label)
+    return np.array(X), np.array(y)
+n_steps = 60
+
+# Preparing the dataset for training
+X_train, y_train = prepare_dataset(df_plvo_train['Sim_Discharge_Standardized'], df_plvo_train['USGS_Discharge_Standardized'], n_steps)
+
+# Preparing the dataset for validation
+X_val, y_val = prepare_dataset(df_plvo_val['Sim_Discharge_Standardized'], df_plvo_val['USGS_Discharge_Standardized'], n_steps)
+
+# Preparing the dataset for testing
+X_test, y_test = prepare_dataset(df_plvo_test['Sim_Discharge_Standardized'], df_plvo_test['USGS_Discharge_Standardized'], n_steps)
+
+X_train.shape, y_train.shape,X_val.shape, y_val.shape,X_test.shape, y_test.shape
+
+print(X_train.shape)
+
+"""**Long Short Term Memory**"""
+
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.layers import *
+from tensorflow.keras.callbacks import ModelCheckpoint, EarlyStopping
+from tensorflow.keras.losses import MeanSquaredError
+from tensorflow.keras.metrics import RootMeanSquaredError
+from tensorflow.keras.optimizers import Adam
+
+model1 = Sequential()
+model1.add(InputLayer(input_shape=(60, 1)))
+model1.add(LSTM(16, activation='relu', return_sequences=True, kernel_regularizer=tf.keras.regularizers.l2(0.0001)))
+model1.add(Dropout(0.35))
+model1.add(LSTM(8, activation='relu', kernel_regularizer=tf.keras.regularizers.l2(0.0001)))
+model1.add(Dropout(0.4))
+model1.add(Dense(8, activation='relu', kernel_regularizer=tf.keras.regularizers.l2(0.0001)))
+model1.add(Dropout(0.35))
+model1.add(Dense(1, activation='linear'))
```

Comment:
If you change the activation function to `relu` you won't ever have any negative predictions
```suggestion
model1.add(Dense(1, activation='relu'))
```